From 8d21b43f569adcd3bb76d3f9d921f2aaddb2c303 Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Fri, 10 Jan 2025 16:18:14 +0100
Subject: [PATCH] Fix build with Qt 6.9

The arg() overload takes double as first parameter, not int
---
 src/lib/marble/geodata/data/GeoDataCoordinates.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/lib/marble/geodata/data/GeoDataCoordinates.cpp b/src/lib/marble/geodata/data/GeoDataCoordinates.cpp
index 6f9d139743..f7bbb79134 100644
--- a/src/lib/marble/geodata/data/GeoDataCoordinates.cpp
+++ b/src/lib/marble/geodata/data/GeoDataCoordinates.cpp
@@ -487,7 +487,7 @@ QString GeoDataCoordinates::lonToString(qreal lon, GeoDataCoordinates::Notation
         if (notation == DMS) {
             // Includes -1 case!
             if (precision < 5) {
-                lonString += QStringLiteral(" %3\"").arg(lonSec, 2, 'f', 0, QLatin1Char('0'));
+                lonString += QStringLiteral(" %3\"").arg((double)lonSec, 2, 'f', 0, QLatin1Char('0'));
                 return lonString + weString;
             }
 
@@ -544,7 +544,7 @@ QString GeoDataCoordinates::lonToString(qreal lon, GeoDataCoordinates::Notation
 
         // Includes -1 case!
         if (precision < 5) {
-            lonString += QStringLiteral(" %3\"").arg(lonSec, 2, 'f', 0, QLatin1Char('0'));
+            lonString += QStringLiteral(" %3\"").arg((double)lonSec, 2, 'f', 0, QLatin1Char('0'));
             return lonString;
         }
 
@@ -649,7 +649,7 @@ QString GeoDataCoordinates::latToString(qreal lat, GeoDataCoordinates::Notation
         if (notation == DMS || notation == Astro) {
             // Includes -1 case!
             if (precision < 5) {
-                latString += QStringLiteral(" %3\"").arg(latSec, 2, 'f', 0, QLatin1Char('0'));
+                latString += QStringLiteral(" %3\"").arg((double)latSec, 2, 'f', 0, QLatin1Char('0'));
                 return latString + nsString;
             }
 
-- 
GitLab

From a14a3a911f5a8f152783a97410267a6fd98cce48 Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Fri, 10 Jan 2025 16:17:36 +0100
Subject: [PATCH] Don't try to compare a QString with enum

I can only speculate about the intention of this code, but it seems clear enough that we want to check the result here
---
 src/lib/marble/MapWizard.cpp | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/lib/marble/MapWizard.cpp b/src/lib/marble/MapWizard.cpp
index 01a0f545db..36d715c617 100644
--- a/src/lib/marble/MapWizard.cpp
+++ b/src/lib/marble/MapWizard.cpp
@@ -465,22 +465,22 @@ void MapWizard::processImageResults()
 
     if (d->owsManager.imageRequestResult().imageStatus() == WmsImageFailed) {
         QMessageBox::information(this, tr("%1").arg(imageType), tr("The %1 could not be downloaded.").arg(imageType));
-        if (imageType == QChar(PreviewImage))
-            d->m_previewImageValid = false; // PORT_QT6 : comparison between enum and QString???
-        if (imageType == QChar(LevelZeroTile))
+        if (d->owsManager.imageRequestResult().resultType() == PreviewImage)
+            d->m_previewImageValid = false;
+        if (d->owsManager.imageRequestResult().resultType() == LevelZeroTile)
             d->m_levelZeroTileValid = false;
-        if (imageType == QChar(LegendImage))
+        if (d->owsManager.imageRequestResult().resultType() == LegendImage)
             d->m_legendImageValid = false;
     } else if (d->owsManager.imageRequestResult().imageStatus() == WmsImageFailedServerMessage) {
         QMessageBox::information(
             this,
             tr("%1").arg(imageType),
             tr("The %1 could not be downloaded successfully. The server replied:\n\n%2").arg(imageType, QString::fromLatin1(d->owsManager.resultRaw())));
-        if (imageType == QChar(PreviewImage))
+        if (d->owsManager.imageRequestResult().resultType() == PreviewImage)
             d->m_previewImageValid = false;
-        if (imageType == QChar(LevelZeroTile))
+        if (d->owsManager.imageRequestResult().resultType() == LevelZeroTile)
             d->m_levelZeroTileValid = false;
-        if (imageType == QChar(LegendImage))
+        if (d->owsManager.imageRequestResult().resultType() == LegendImage)
             d->m_legendImageValid = false;
     } else {
         if (d->owsManager.imageRequestResult().resultType() == PreviewImage) {
-- 
GitLab

